frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices-mode/install-wizard-steps/storage-and-nodes-step.tsx

```diff
@@ -81,34 +81,62 @@ export const StorageAndNodes: React.FC<StorageAndNodesProps> = ({ state, dispatc
     storageClass,
     storageClassName: scName,
     enableMinimal,
-    nodes,
     stretchClusterChecked,
     enableFlexibleScaling,
+    chartNodes = new Set(),
   } = state;
 
-  const scNodeNames: React.MutableRefObject<string[]> = React.useRef([]); // names of the nodes, backing the storage of selected storage class
+  const memoizedPvData = useDeepCompareMemoize(pvData, true);
+  const memoizedNodesData = useDeepCompareMemoize(nodesData, true);
 
-  const { cpu, memory, zones } = getNodeInfo(nodes);
-  const nodesCount: number = nodes.length;
+  const pvs: K8sResourceKind[] = React.useMemo(() => getSCAvailablePVs(memoizedPvData, scName), [
+    memoizedPvData,
+    scName,
+  ]);
+  const associatedNodes = getAssociatedNodes(pvs);
+  const tableData: NodeKindWithLoading[] =
+    associatedNodes?.length > 0 && pvLoaded && nodesLoaded
+      ? memoizedNodesData
+          ?.filter((node) => chartNodes.has(getName(node)))
```

Comment:
Ah! I mentioned adding this but it's not helping with this implementation.
We can do `lvsIsSelectNodes ? lvsSelectedNodes: lvsAllNodes` to cover all nodes with/without disks.
(sorry for that)